Analysis/Tasks/PWGCF/dptdptcorrelations.cxx

```diff
@@ -0,0 +1,407 @@
+{
+using CollisionEvSelCent = soa::Join<aod::Collisions, aod::EvSels, aod::Cents>::iterator;
+using TrackData = soa::Join<aod::Tracks, aod::TracksCov, aod::TracksExtra, aod::TracksExtended, aod::TrackSelection>::iterator;
+
+/* we have to change from int to bool when bool columns work properly */
+namespace dptdptcorrelations
+{
+DECLARE_SOA_COLUMN(EventAccepted, eventaccepted, uint8_t);
+DECLARE_SOA_COLUMN(TrackacceptedAsOne, trackacceptedasone, uint8_t);
+DECLARE_SOA_COLUMN(TrackacceptedAsTwo, trackacceptedastwo, uint8_t);
+} // namespace dptdptcorrelations
+DECLARE_SOA_TABLE(AcceptedEvents, "AOD", "ACCEPTEDEVENTS", dptdptcorrelations::EventAccepted);
+DECLARE_SOA_TABLE(ScannedTracks, "AOD", "SCANNEDTRACKS", dptdptcorrelations::TrackacceptedAsOne, dptdptcorrelations::TrackacceptedAsTwo);
+} // namespace aod
+} // namespace o2
+
+namespace dptdptcorrelations
+{
+/* all this bins have to be make configurable */
+int ptbins = 18;
+float ptlow = 0.2, ptup = 2.0;
+int etabins = 16;
+float etalow = -0.8, etaup = 0.8;
+int zvtxbins = 40;
+float zvtxlow = -10.0, zvtxup = 10.0;
+int phibins = 72;
+std::string fTaskConfigurationString = "PendingToConfigure";
+
+Configurable<int> cfgTrackType{"trktype", 1, "Type of selected tracks: 0 = no selection, 1 = global tracks loose DCA, 2 = global SDD tracks"};
+Configurable<int> cfgTrackOneCharge("trk1charge", 1, "Trakc one charge: 1 = positive, -1 = negative");
+Configurable<int> cfgTrackTwoCharge("trk2charge", -1, "Trakc two charge: 1 = positive, -1 = negative");
+Configurable<bool> cfgProcessPairs("processpairs", true, "Process pairs: false = no, just singles, true = yes, process pairs");
+
+/* while we don't have proper bool columnst */
+uint8_t DPTDPT_TRUE = 1;
+uint8_t DPTDPT_FALSE = 0;
+
+/// \enum SystemType
+/// \brief The type of the system under analysis
+enum SystemType {
+  kNoSystem = 0, ///< no system defined
+  kpp,           ///< **p-p** system
+  kpPb,          ///< **p-Pb** system
+  kPbp,          ///< **Pb-p** system
+  kPbPb,         ///< **Pb-Pb** system
+  kXeXe,         ///< **Xe-Xe** system
+  knSystems      ///< number of handled systems
+};
+
+/* probably this will not work in the context of distributed processes */
+/* we need to revisit it!!!!                                           */
+SystemType fSystem = kNoSystem;
+TH1F* fhCentMultB = nullptr;
+TH1F* fhCentMultA = nullptr;
+TH1F* fhVertexZB = nullptr;
+TH1F* fhVertexZA = nullptr;
+TH1F* fhPtB = nullptr;
+TH1F* fhPtA = nullptr;
+TH1F* fhPtPosB = nullptr;
+TH1F* fhPtPosA = nullptr;
+TH1F* fhPtNegB = nullptr;
+TH1F* fhPtNegA = nullptr;
+
+TH1F* fhEtaB = nullptr;
+TH1F* fhEtaA = nullptr;
+
+TH1F* fhPhiB = nullptr;
+TH1F* fhPhiA = nullptr;
+
+TH2F* fhEtaVsPhiB = nullptr;
+TH2F* fhEtaVsPhiA = nullptr;
+
+TH2F* fhPtVsEtaB = nullptr;
+TH2F* fhPtVsEtaA = nullptr;
+
+SystemType getSystemType()
+{
+  /* we have to figure out how extract the system type */
+  return kPbPb;
+}
+
+bool IsEvtSelected(aod::CollisionEvSelCent const& collision)
```

Comment:
This can definitely be implemented as a filter.

Reply:
I agree